spec.html

```diff
@@ -181,19 +181,29 @@ <h1>RealmImportValue ( _specifierString_, _exportNameString_, _callerRealm_, _ev
 				1. Assert: Type(_exportNameString_) is String.
 				1. Assert: _callerRealm_ is a Realm Record.
 				1. Assert: _evalRealm_ is a Realm Record.
-				1. Let _promiseCapability_ be ! NewPromiseCapability(%Promise%).
+				1. Let _innerCapability_ be ! NewPromiseCapability(%Promise%).
 				1. Let _runningContext_ be the running execution context.
 				1. If _runningContext_ is not already suspended, suspend _runningContext_.
 				1. Push _evalContext_ onto the execution context stack; _evalContext_ is now the running execution context.
-				1. Perform ! HostImportModuleBindingDynamically(_callerRealm_, _evalRealm_, _specifierString_, _exportNameString_, _promiseCapability_).
+				1. Perform ! HostImportModuleDynamically(*null*, _specifierString_, _innerCapability_).
 				1. Suspend _evalContext_ and remove it from the execution context stack.
 				1. Resume the context that is now on the top of the execution context stack as the running execution context.
-				1. Return _promiseCapability_.[[Promise]].
+				1. Let _steps_ be the steps of an ExportGetter function as described below.
+				1. Let _onFulfilled_ be ! CreateBuiltinFunction(_steps_, 1, "", « [[ExportNameString]] », _callerRealm_).
+				1. Set _onFulfilled_.[[ExportNameString]] to _exportNameString_.
+				1. Let _promiseCapability_ be ! NewPromiseCapability(%Promise%).
+				1. Return ! PerformPromiseThen(_innerCapability_.[[Promise]], _onFulfilled_, _callerRealm_.[[Intrinsics]].[[%ThrowTypeError%]], _promiseCapability_).
```

Comment:
I'm not sure it should be a type error, in the previous version I was more hand waving (intentionally). When you fail to wrap, the error is not a type error either, isn't it?

Reply:
```js
import('bla').then(() => console.log('noop')).catch(console.log);
// TypeError: Failed to resolve module specifier 'bla'
//    at <anonymous>:1:1
```

In general this would also wrap import errors into TypeError (if module 'bla' exists and throws an error on evaluation).